McZapkie/Mover.cpp

```diff
@@ -147,20 +154,30 @@ void TSecuritySystem::update(double dt, double vel, bool pwr) {
 
 	if (vigilance_timer > AwareDelay
 	        || press_timer > MaxHoldTime
-	        || cabsignal_active)
+	        || cabsignal_active) {
 		alert_timer += dt;
+    }
+
+    /* zeby rozroznic ca od shp */
+    if (vigilance_timer > AwareDelay || press_timer > MaxHoldTime) {
+        alerter_active = true;
+    }
 }
 
 void TSecuritySystem::set_cabsignal() {
     if (cabsignal_enabled && power)
 		cabsignal_active = true;
 }
 
+bool TSecuritySystem::has_separate_acknowledge() const {
+	return separate_acknowledge;
+}
+
 bool TSecuritySystem::is_blinking() const {
     if (!power)
         return false;
 
-	return alert_timer > 0.0;
+	return separate_acknowledge ? alerter_active : (alert_timer > 0.0);
```

Comment:
I'm not convinced why `alerter_active` and this change is necessary. There's already `is_vigilance_blinking` and `is_cabsignal_blinking`. In case some other part of the code is incorrectly using `is_blinking` in place of `is_vigilance_blinking` it should be changed there, instead of special-casing `is_blinking`.